board/rt1051/memcpy-armv7m.S

```diff
@@ -0,0 +1,329 @@
+/*
+ * Copyright (c) 2013 ARM Ltd
```

Comment:
Instead of this assembly version, I'd like to suggest using a plain C version from here: 
https://github.com/travisg/newos/blob/master/lib/libc/string/memcpy.c
It's also used in [Android](https://android.googlesource.com/kernel/lk/+/master/lib/libc/string/memcpy.c).
It's short, readable and optimized for handling aligned and unaligned addresses.
And it compiles to [fewer instructions](https://gcc.godbolt.org/z/n4r1jdc46) than this one.